server/ai/types.ts

```diff
@@ -120,6 +120,11 @@ export const SearchAnswerResponse = z.object({
   usefulIndex: z.array(z.number()),
 })
 
+export const ToolAnswerResponse = z.object({
```

Comment:
How can we allow this type to be extendable, like for `arguments` we could get other types to extend this type to make it specific to that App. We want to be able to go specific types if we know which App + Tool we are dealing with for stricter argument parsing. 


first we make the type
```js
export type ToolAnswerResponse = z.infer<typeof ToolAnswerResponseSchema>; // we could name it ToolResponse as well


export const GitHubCreateIssueResponseSchema =
  ToolAnswerResponseSchema.extend({
    tool: z.literal("github.createIssue"),
    arguments: GitHubCreateIssueArgs,
  });
  ```